arviz/utils.py

```diff
@@ -50,14 +51,26 @@ def _var_names(var_names, data):
                 )
             )
 
+        excluded_vars = [
+            var[1:] for var in var_names if var.startswith("~") and var not in all_vars
+        ]
         if excluded_vars:
-            var_names = [i for i in all_vars if i not in excluded_vars]
+            if filter_like:
+                for var in excluded_vars:
+                    if var not in all_vars:
+                        real_vars = [real_var for real_var in all_vars if var in real_var]
+                        excluded_vars.extend(real_vars)
+                        excluded_vars.remove(var)
```

Comment:
what would happen if one of the `real_vars` was the same as `var`? i.e. `theta` in eight schools -> exlcude `theta` and `theta_t`

Reply:
Mmmh I don't think it's possible because `real_var` is in `all_vars`, but `var` is not in `all_vars`. Is that your question?

Reply:
Are we on the same page here @OriolAbril ?